### PR TITLE
Toolkit: Auth - Passport - Allow custom strategy name

### PR DESCRIPTION
**Pull request by noxify**
Closes: #3686 

### What are the changes and their implications?

* Adds a new optional property (`name`) to the Passport Strategy definition
* Checks if there is a strategy for the manual specified name, if not, it will use the default strategy name
* Added a new example app to test the changes

## Feature Checklist

- [x] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed) — an example app was added
- [x] Documentation added/updated (submit PR to [blitzjs.com repo `canary` branch](https://github.com/blitz-js/blitzjs.com/tree/canary)): https://github.com/blitz-js/blitzjs.com/pull/747

## Miscellaneous 

I have tested it with the following packages:
* Strategy: https://github.com/panva/node-openid-client
* Local Mock Server: https://github.com/axa-group/oauth2-mock-server

### CODE REVIEW

**Comment by dillonraphael** [comment]
Good job @noxify! Give us a little bit of time to review this. Thanks for also adding the passport example app.

**Comment by noxify** [comment]
@dillonraphael - take your time :)

File: packages/blitz-auth/src/server/passport-adapter.ts [hunk 1]
```diff
@@ -108,15 +109,18 @@ export function passportAuth(config: BlitzPassportConfig): ApiHandler {
       "No Passport strategies found! Please add at least one strategy.",
     )
 
-    const blitzStrategy = configObject.strategies.find(
-      ({strategy}) => strategy.name === req.query?.auth?.[0] ?? "",
+    // find the requested strategy based on the manual specified name
+    // or if name is not specified, based on the strategy name
```
Comment (beerose) [inline packages/blitz-auth/src/server/passport-adapter.ts:113]:
```suggestion
    // Find the requested strategy based on the manually specified name.
    // Use default strategy name if a custom name is not specified.
```

File: packages/blitz-auth/src/server/passport-adapter.ts [hunk 2]
```diff
@@ -108,15 +109,18 @@ export function passportAuth(config: BlitzPassportConfig): ApiHandler {
...
     )
     assert(blitzStrategy, `A passport strategy was not found for: ${req.query.auth[0]}`)
 
-    const {strategy, authenticateOptions} = blitzStrategy
+    const {name, strategy, authenticateOptions} = blitzStrategy
 
-    passport.use(strategy)
-    const strategyName = strategy.name as string
+    const strategyName = (name || strategy.name) as string
+
```
Comment (beerose) [inline packages/blitz-auth/src/server/passport-adapter.ts:122]: Let's add this assertion to get rid of `as string` when assigning `strategyName`:

```suggestion
    assert(typeof strategy.name  !== "undefined", `A passport strategy name was not found for: ${req.query.auth[0]}`)
    
    const strategyName = name || strategy.name
```

**Review by beerose — COMMENTED** [review]
Looks and works great!

Left two minor comments

**Comment by noxify** [comment]
@beerose / @dillonraphael - commited the suggested changes.

I hope that I can find some time in the next days to create also the PR for the docs.

**Comment by beerose** [comment]
> @beerose / @dillonraphael - commited the suggested changes.
> 
> I hope that I can find some time in the next days to create also the PR for the docs.

Thanks for making the changes! I took the liberty of creating a docs PR for this :)

**Comment by beerose** [comment]
There are three remaining things to do (I tried to do them myself but can't push to your remote):

1. Add changeset — you need to run `pnpm changeset` in root and then select `@blitzjs/auth` with a spacebar. The changeset message could be `Allow specifying custom strategy name in Blitz's passport adapter`.
2. Add `"toolkit-app-passportjs": "1.0.0",` to `.changeset/pre.json`
3. Update `.changeset/config.json` to have this: `"ignore": ["web", "test-*", "toolkit-*"]` for the `ignore` property.

**Comment by noxify** [comment]
@beerose - done - if there is something else, please let me know

**Comment by noxify** [comment]
since i will be only partly available until next monday, i added you ( beerose ) as contributor to my fork.

File: apps/toolkit-app-passportjs/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+## 1.0.1-alpha.0
+
+### Patch Changes
+
+- Updated dependencies
+  - blitz@2.0.0-alpha.5
+  - @blitzjs/auth@2.0.0-alpha.5
+  - @blitzjs/next@2.0.0-alpha.5
+  - @blitzjs/rpc@2.0.0-alpha.5
+  - @blitzjs/config@2.0.0-alpha.5
```
Comment (beerose) [inline apps/toolkit-app-passportjs/CHANGELOG.md:185]:
```suggestion
```

File: apps/toolkit-app-passportjs/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
+# toolkit-app
```
Comment (beerose) [inline apps/toolkit-app-passportjs/CHANGELOG.md:1]:
```suggestion
# toolkit-app-passportjs
```

**Review by beerose — APPROVED** [review]

**Comment by noxify** [comment]
Thanks @beerose ❤️

**Pull request merged**